ajax/timer.php

```diff
@@ -12,79 +12,128 @@
          break;
 
       case 'end':
+      case 'pause':
          if (PluginActualtimeTask::checkTimerActive($task_id)) {
+
             if (PluginActualtimeTask::checkUser($task_id, Session::getLoginUserID())) {
+
                $actual_begin=PluginActualtimeTask::getActualBegin($task_id);
                $seconds=(strtotime(date("Y-m-d H:i:s"))-strtotime($actual_begin));
                $DB->update(
                   'glpi_plugin_actualtime_tasks', [
-                     'actual_end'      => date("Y-m-d H:i:s"),
-                     'actual_actiontime'      => $seconds,
+                     'actual_end'        => date("Y-m-d H:i:s"),
+                     'actual_actiontime' => $seconds,
                   ], [
-                     'tasks_id'=>$task_id,
+                     'tasks_id' => $task_id,
                      [
                         'NOT' => ['actual_begin' => null],
                      ],
-                     'actual_end'=>null,
+                     'actual_end' => null,
                   ]
                );
-               $DB->update(
-                  'glpi_tickettasks', [
-                     'state'=>2,
-                  ], [
-                     'id'=>$task_id,
-                  ]
-               );
-
+               if ($_POST['action']=='end') {
+                  $DB->update(
+                     'glpi_tickettasks', [
+                        'state' => 2,
+                     ], [
+                        'id' => $task_id,
+                     ]
+                  );
+               }
                $result=[
-                  'mensage'=>__("Timer completed", 'actualtime'),
-                  'title' => __s('Information'),
-                  'class' => 'info_msg',
+                  'mensage' => __("Timer completed", 'actualtime'),
+                  'title'   => __('Information'),
+                  'class'   => 'info_msg',
                ];
+
             } else {
+
                $result=[
-                  'mensage'=>__("Only the user who initiated the task can close it", 'actualtime'),
-                  'title' => __s('Warning'),
-                  'class' => 'warn_msg',
+                  'mensage' => __("Only the user who initiated the task can close it", 'actualtime'),
+                  'title'   => __('Warning'),
+                  'class'   => 'warn_msg',
                ];
+
             }
+
          } else {
-            $result=[
-               'mensage'=>__("The task had not been initialized", 'actualtime'),
-               'title' => __s('Warning'),
-               'class' => 'warn_msg',
-            ];
+
+            if ($_POST['action']=='pause') {
+               $result=[
+                  'mensage' => __("The task had not been initialized", 'actualtime'),
+                  'title'   => __('Warning'),
+                  'class'   => 'warn_msg',
+               ];
+            } else {
+
+               if (PluginActualtimeTask::checkTech($task_id)) {
```

Comment:
This check is redundant.
It is already checked when you show the buttons
Please remove it

Reply:
Yes, it is redundant, as stated in the code itself. I was just cautious if someone could use the ajax resquest to do something wrong, so double checked any database modification. Anyway, you are right. It is just checking twice. Branch was corrected with commit (https://github.com/Multiconecta/actualtime/commit/4e60eb8dc1288a732df7bb50d8c91a6a0ea236e5)